src/main/java/com/indeed/jiraactions/jiraissues/JiraIssuesProcess.java

```diff
@@ -0,0 +1,139 @@
+package com.indeed.jiraactions.jiraissues;
+
+import org.joda.time.DateTime;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+
+public class JiraIssuesProcess {
+    private static final Logger log = LoggerFactory.getLogger(JiraIssuesIndexBuilder.class);
+
+    private final List<Map<String, String>> newIssuesMapped = new ArrayList<>();
+    private final List<String> nonApiStatuses = new ArrayList<>(); // Old statuses that don't show up in the API.
+    private List<String[]> newIssues;
+    private List<String> newFields; // Fields from jiraaction's updated issues.
+    private List<String> oldFields; // Fields from previous TSV
+
+    public final DateTime startDate;
+
+    public JiraIssuesProcess(final DateTime startDate) {
+        this.startDate = startDate;
+    }
+
+    public void convertToMap() {
+        for(int issue = 1; issue < newIssues.size(); issue++) {
+            final Map<String, String> mappedLine = new LinkedHashMap<>(newFields.size());
+            final String[] line = newIssues.get(issue);
+            if(line == null) {
+                break;
+            } else {
+                for(int i = 0; i < line.length; i++) {
+                    mappedLine.put(newFields.get(i), line[i]);
+                }
+                newIssuesMapped.add(mappedLine);
+            }
+        }
+    }
+
+    /* If the issue is updated through jiraactions it will replace it because that version is the latest.
+     * If it isn't replaced then it gets updated -- only fields involving time are updated so this is really easy.
+     * Issues from jiraactions are removed when they get replaced meaning that the ones remaining are new issues and are therefore added.
+     */
+    public Map<String, String> compareAndUpdate(final String[] issue) {
+        final int filter = Integer.parseInt(startDate.minusMonths(6).toString("yyyyMMdd"));
```

Comment:
This filter should also be configurable.

Reply:
Also this is another thing that seems like a good candidate to be a member variable, rather than local to this method.